src/sentry/tasks/auth.py

```diff
@@ -20,39 +21,47 @@
 
 
 @instrumented_task(name="sentry.tasks.send_sso_link_emails", queue="auth")
-def email_missing_links(org_id, actor_id, provider_key, **kwargs):
+def email_missing_links(org_id: int, actor_id: int, provider_key: str, **kwargs):
     try:
         org = Organization.objects.get(id=org_id)
-        actor = User.objects.get(id=actor_id)
         provider = manager.get(provider_key)
-    except (Organization.DoesNotExist, User.DoesNotExist, ProviderNotRegistered) as e:
+    except (Organization.DoesNotExist, ProviderNotRegistered) as e:
         logger.warning("Could not send SSO link emails: %s", e)
         return
 
+    user = user_service.get_user(user_id=actor_id)
+    if not user:
+        logger.warning(f"Could not send SSO link emails. Unable to find user by id: {actor_id}")
```

Comment:
Could we put the `actor_id` into the `extra` parameter? Our logging infrastructure doesn't hand high cardinality messages well. Something like:

```suggestion
        logger.warning(f"sso.link.email_failure.could_not_find_user", extra={"user_id": actor_id})
```
would be ok though.